fix(docs): Update `README` Components links to new URL pattern

tulup-conner opened the pull request:
We don't use `HashRouter` anymore, so the links no longer begin with `#`. Also adds anchors to the `Forms` page for each of the components listed (radio, checkbox, and so forth).

## Breaking changes

None.

## Features

None.

## Bug fixes

- [x] Update `README` Components links to new URL pattern

bacali95 commented on README.md:
```diff
@@ -18,7 +18,7 @@
       <img alt="Latest release" src="https://badge.fury.io/js/flowbite-react.svg">
     </a>
-    <a href="https://flowbite.com/docs/getting-started/license/">
+    <a href="https://flowbite-react.com/getting-started/license/">
```
This one needs to be reverted

bacali95 commented on README.md:
```diff
@@ -50,7 +50,7 @@ Documentation for `flowbite-react` is not yet finished.
 If you want to browse the components, visit [flowbite-react.com](https://flowbite-react.com/).
 
-If you want to learn more about Flowbite, visit [Flowbite docs](https://flowbite.comhttps://flowbite.com/docs/getting-started/introduction/).
+If you want to learn more about Flowbite, visit [Flowbite docs](https://flowbite-react.comhttps://flowbite-react.com/getting-started/introduction/).
```
The link seems to be broken can you please fi it

bacali95 commented on README.md:
```diff
@@ -373,10 +373,10 @@ Thank you for your interest in helping! Visit our [guide on contributing](https:
 If you need the Figma files for the components you can check out our website for more information:
 
-🎨 [Get access to the Figma design files](https://flowbite.com/figma/)
+🎨 [Get access to the Figma design files](https://flowbite-react.com/figma/)
```
This one should be reverted

bacali95 commented on README.md:
```diff
@@ -373,10 +373,10 @@ Thank you for your interest in helping! Visit our [guide on contributing](https:
 The Flowbite name and logos are trademarks of Crafty Dwarf Inc.
 
-📝 [Read about the licensing terms](https://flowbite.com/docs/getting-started/license/)
+📝 [Read about the licensing terms](https://flowbite-react.com/getting-started/license/)
```
This one as well

tulup-conner commented:
Thank you! I made those changes and also:

- [x] Changed `Sidebar` to not be under construction
- [x] Reverted some other links that were broken (Typography, components that are not yet converted to `flowbite-react` yet)

bacali95 commented:
> @bacali95 
> 
> Thank you! I made those changes and also:
> 
> - [x] Changed `Sidebar` to not be under construction
> - [x] Reverted some other links that were broken (Typography, components that are not yet converted to `flowbite-react` yet)

Great thanks, let's merge this one
